taca/illumina/__init__.py

```diff
@@ -158,9 +67,41 @@ def is_finished(self):
         """
         return os.path.exists(os.path.join(self.run_dir, 'RTAComplete.txt'))
 
+
+    def demultiplex(self):
+        """Perform demultiplexing of the flowcell.
+
+        Takes software (bcl2fastq version to use) and parameters from the configuration
+        file.
+        """
+        logger.info('Building bcl2fastq command')
+        config = CONFIG['analysis']
+        with chdir(self.run_dir):
+            cl = [config.get('bcl2fastq').get(self.run_type)]
+            if config['bcl2fastq'].has_key('options'):
+                cl_options = config['bcl2fastq']['options']
+
+                # Append all options that appear in the configuration file to the main command.
+                # Options that require a value, i.e --use-bases-mask Y8,I8,Y8, will be returned
+                # as a dictionary, while options that doesn't require a value, i.e --no-lane-splitting
+                # will be returned as a simple string
+                for option in cl_options:
+                    if isinstance(option, dict):
+                        opt, val = option.popitem()
+                        cl.extend(['--{}'.format(opt), str(val)])
+                    else:
+                        cl.append('--{}'.format(option))
+
+            logger.info(("BCL to FASTQ conversion and demultiplexing started for "
+                      " run {} on {}".format(os.path.basename(self.id), datetime.now())))
```

Comment:
![you're](http://global3.memecdn.com/im-sorry-son-but-were-grammar-nazis_o_823267.jpg)